core/db/migrate/20201108142424_add_deprecated_to_payment_method.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddDeprecatedToPaymentMethod < ActiveRecord::Migration[6.0]
+  def change
+    add_column :spree_payment_methods, :deprecated, :boolean, default: false
```

Comment:
I don't think "deprecated" is the right name for this field. "Deprecated" means "usable but regarded as obsolete and best avoided, typically due to having been superseded". I think in this case the payment method will no longer be usable, so we should find a better way to describe it.

Reply:
As I said discussing with @spaghetticode, I'd like the db row to be clearly marked as "deprecated" (or another term). I didn't understand from the solidus docs in which cases should a payment method be flagged with `active = false`, and nor if there are cases where it makes sense to set it to `false` at some point and again to `true` after a while. So, I wasn't really sure that the `active` could be used for this case.

Given that, if you think that `active` can be used to express this state I would be glad to remove the new flag, or even better to change it into `previous_type` as suggested from @spaghetticode to keep track of the previous real type of the payment method.

Reply:
These concerns make sense and I don't have a clear answer as well. Just thinking aloud: is there any occurrence when we want `active` and `deprecated` to have different values at the same time. I mean does it make sense having `deprecated: true` without also having `active: false`? I don't think so, since the payment method class has been removed from the project, having that payment method active would just produce this error more often. This makes me think that active is just fine for this purpose as well. 

If we also add `previous_type`, this new field populated would be enough to understand which records are in this specific state.

But again, I don't have strong opinions here and I'd leave this decision to you. I'm fine with both as soon as we don't use the `deprecated` term for the reasons above.

Reply:
Then adding `previous_type` and removing `deprecated` seems a good choice. I'm just not 100% sure on the name, can you think of something more expressive and not too long? Maybe `removed_type`?

Reply:
I changed `deprecated` to `previous_type` and updated the doc accordingly.